modules/secure-cloud-run-core/variables.tf

```diff
@@ -151,7 +151,161 @@ variable "env_vars" {
 }
 
 variable "members" {
-  type        = list(string)
   description = "Users/SAs to be given invoker access to the service with the prefix `serviceAccount:' for SAs and `user:` for users."
+  type        = list(string)
   default     = []
 }
+
+variable "generate_revision_name" {
+  description = "Option to enable revision name generation."
+  type        = bool
+  default     = true
+}
+
+variable "traffic_split" {
+  description = "Managing traffic routing to the service."
+  type = list(object({
+    latest_revision = bool
+    percent         = number
+    revision_name   = string
+  }))
+  default = [{
+    latest_revision = true
+    percent         = 100
+    revision_name   = "v1-0-0"
+  }]
+}
+
+variable "service_labels" {
+  description = "A set of key/value label pairs to assign to the service."
+  type        = map(string)
+  default     = {}
+}
+
+// Metadata
+variable "template_labels" {
+  description = "A set of key/value label pairs to assign to the container metadata."
+  type        = map(string)
+  default     = {}
+}
+
+// template spec
+variable "container_concurrency" {
+  description = "Concurrent request limits to the service."
+  type        = number
+  default     = null
+}
+
+variable "timeout_seconds" {
+  description = "Timeout for each request."
+  type        = number
+  default     = 120
+}
+
+variable "service_account_email" {
+  description = "Service Account email needed for the service."
+  type        = string
+  default     = ""
+}
+
+variable "volumes" {
+  description = "[Beta] Volumes needed for environment variables (when using secret)."
+  type = list(object({
+    name = string
+    secret = set(object({
+      secret_name = string
+      items       = map(string)
+    }))
+  }))
+  default = []
+}
+
+# template spec container
+# resources
+# cpu = (core count * 1000)m
+# memory = (size) in Mi/Gi
+variable "limits" {
+  description = "Resource limits to the container."
+  type        = map(string)
+  default     = null
+}
+variable "requests" {
+  description = "Resource requests to the container."
+  type        = map(string)
+  default     = {}
+}
+
+variable "ports" {
+  description = "Port which the container listens to (http1 or h2c)."
+  type = object({
+    name = string
+    port = number
+  })
+  default = {
+    name = "http1"
+    port = 8080
+  }
+}
+
+variable "argument" {
+  description = "Arguments passed to the ENTRYPOINT command, include these only if image entrypoint needs arguments."
+  type        = list(string)
+  default     = []
+}
+
+variable "container_command" {
+  description = "Leave blank to use the ENTRYPOINT command defined in the container image, include these only if image entrypoint should be overwritten."
+  type        = list(string)
+  default     = []
+}
+
+variable "env_secret_vars" {
```

Comment:
We should probably not surface the variable to pass in secrets as environmental variables as our security best practice will be to use volume mounts and we want to be very opinionated in this blueprint.

Reply:
Good catch! It's fixed now.